src/utilities/icon-utils.test.ts

```diff
@@ -0,0 +1,300 @@
+            // Arrange
+            IconUtils.registerSvgIcon(getSvgIconByType(Icons.ChevronDown));
+
+            // Act & Assert
+            expect(IconUtils.getSvg(Icons.Checkmark)).toBeUndefined();
+        });
+
+        test("given match in registry, when no size provided, returns base Svg", () => {
+            // Arrange
+            const ChevronDown = getSvgIconByType(Icons.ChevronDown);
+            IconUtils.registerSvgIcon(ChevronDown);
+
+            // Act
+            const result = IconUtils.getSvg(Icons.ChevronDown);
+
+            // Assert
+            expect(result).toBe(ChevronDown.base);
+        });
+
+        test.each`
+            size
+            ${IconSizes.Base}
+            ${IconSizes.Large}
+        `(
+            "given match in registry, when size of '$size' provided, returns correct Svg",
+            ({ size }) => {
+                // Arrange
+                const ChevronDown = getSvgIconByType(Icons.ChevronDown);
+                IconUtils.registerSvgIcon(ChevronDown);
+
+                // Act
+                const result = IconUtils.getSvg(Icons.ChevronDown, size);
+
+                // Assert
+                expect(result).toBe(ChevronDown[size]);
+            }
+        );
+    });
+
+    // #endregion getSvg
+
+    // -----------------------------------------------------------------------------------------
+    // #region getSvgIcon
+    // -----------------------------------------------------------------------------------------
+
+    describe("getSvgIcon", () => {
+        test("given no icons registered, returns undefined", () => {
+            expect(IconUtils.getSvgIcon(Icons.Checkmark)).toBeUndefined();
+        });
+
+        test("given no match in registry, returns undefined", () => {
+            // Arrange
+            IconUtils.registerSvgIcon(getSvgIconByType(Icons.ChevronDown));
+
+            // Act & Assert
+            expect(IconUtils.getSvgIcon(Icons.Checkmark)).toBeUndefined();
+        });
+
+        test("given match in registry, returns SvgIcon", () => {
+            // Arrange
+            const ChevronDown = getSvgIconByType(Icons.ChevronDown);
+            IconUtils.registerSvgIcon(ChevronDown);
+
+            // Act
+            const result = IconUtils.getSvgIcon(Icons.ChevronDown);
+
+            // Assert
+            expect(result.type).toBe(ChevronDown.type);
+        });
+    });
+
+    // #endregion getSvgIcon
+
+    // -----------------------------------------------------------------------------------------
+    // #region getRegistry
+    // -----------------------------------------------------------------------------------------
+
+    describe("getRegistry", () => {
+        test("given no icons registered, returns empty object", () => {
+            // Arrange & Act
+            const result = IconUtils.getRegistry();
+
+            // Assert
+            expect(Object.keys(result)).toHaveLength(0);
```

Comment:
Nitpick - you might be able to get away with `expect(result).toBeEmpty()` or at least simplify this with `expect(result).toBe({});`